src/relayer/Relayer.ts

```diff
@@ -363,25 +364,50 @@ export class Relayer {
       const { relayerFeePct, gasCost, gasLimit: _gasLimit, lpFeePct: realizedLpFeePct } = repaymentChainProfitability;
       if (isDefined(repaymentChainId)) {
         const gasLimit = isMessageEmpty(resolveDepositMessage(deposit)) ? undefined : _gasLimit;
-        this.fillRelay(deposit, repaymentChainId, realizedLpFeePct, gasLimit);
+        this.fillRelay(deposit, isLiteChain ? originChainId : repaymentChainId, realizedLpFeePct, gasLimit);
 
         // Update local balance to account for the enqueued fill.
         tokenClient.decrementLocalBalance(destinationChainId, deposit.outputToken, deposit.outputAmount);
       } else {
         profitClient.captureUnprofitableFill(deposit, realizedLpFeePct, relayerFeePct, gasCost);
       }
     } else if (selfRelay) {
```

Comment:
This code is almost never used, so I'd prefer to keep it really small by just updating the existing logic as follows:

```suggestion
    } else if (selfRelay && !isLiteChain) {
```

Reply:
Good call. I kept the internal conditional just because then we'd have to add an else-if to the else statement below.